CONTRIBUTING.md

```diff
@@ -210,6 +210,124 @@ the "actual" output, then re-run `cargo test` and make sure everything passes.
 
 Congrats on the new lint!
 
+#### Adding a witness
+
+**Witnesses** are a new, unstable feature of `cargo-semver-checks` that let us create witness
+code, a minimal compilable example of potential downstream breakage\. They are configured via the
+`witness` field in the lint file `src/lints/<lint_name>.ron`:
+
+If it is `None` (or the field is omitted entirely), `cargo-semver-checks` will not be able
+to generate witness code for this lint\. This can be the right choice, sometimes:
+for example, if this lint is a warn- or allow-by-default lint that hints at potential
+breakage, but won't cause breaking changes directly\. Additionally, if it's not currently
+possible to write a Trustfall query that gets the necessary information to generate
+witnesses, leave this field as `None`, but leave a `// TODO` comment explaining
+what would unblock this lint from being able to generate a witness.
+
+##### Hint templates
+
+When the `witness` field is not `None`, it must have the `hint_template` field\. This is a
+`handlebars` template that generates a small (1-3 line) human-readable message that
+explains the idea of how downstream code would break\. For example, for the `function_missing` lint:
+
+```rust
+use {{join "::" path};
+{{name}}(...);
+```
+
+which could render to something like:
+
+```ron
+witness: (
+  hint_template: r#"use function_missing::will_be_removed_fn;\n\
+  will_be_removed_fn(...);"#
+)
+```
```

Comment:
It might be worth illustrating multi-line templates in a separate example, after going through the basic example end-to-end first.